npm_and_yarn/lib/dependabot/npm_and_yarn/dependency_files_filterer.rb

```diff
@@ -34,6 +38,14 @@ def package_files_requiring_update
 
       attr_reader :dependency_files, :updated_dependencies
 
+      def fetch_paths_requiring_update_check
+        return [File.dirname(root_lockfile.name)] if lockfiles == [root_lockfile]
```

Comment:
```suggestion
        # if a root lockfile exists, it tracks all dependencies
        return [File.dirname(root_lockfile.name)] if lockfiles == [root_lockfile]
```

Reply:
I'll add a comment, but I'll use "if ONLY a lockfile exists, it tracks all dependencies" to make it a bit more clear.